synchronizations/Synchronization.md

```diff
@@ -0,0 +1,223 @@
+            personRootModel.RootElements.Add(personRegister);
+
+            repository.Models.Add(new Uri("ttc:source"), familyRootModel);
+            repository.Models.Add(new Uri("ttc:target"), personRootModel);
+        }
+
+        static void Main(string[] args)
+        {}
+    }
+```
+
+## Synchronization
+
+As described above, in this section, we specifically address the synchronization rule/block.
+
+```csharp 
+synchronization.SynchronizationRule<Families2PersonsSynchronization.FamilyRegisterToPersonRegister>()
+```
+
+The line signifies that a synchronization rule is being invoked, with its type being `Families2PersonsSynchronization.FamilyRegisterToPersonRegister`. This rule leverages the synchronization block, which is defined within the Families2PersonsSynchronization class.
+
+Lets take a closer a look at the synchronization block. 
+```csharp
+public class FamilyRegisterToPersonRegister : SynchronizationRule<FamilyRegister, PersonRegister>
+        {
+            // Synchronization Block
+            public override void DeclareSynchronization()
+            {
+                SynchronizeMany(SyncRule<MemberToMember>(),
+                    fam => new FamilyMemberCollection(fam),
+                    persons => persons.Persons);
+            }
+        }
+```
+This method named [SynchronizeMany](../synchronizations/api/NMF.Synchronizations.SynchronizationRuleBase.yml) facilitates bidirectional synchronization, implying that the method itself defines a dependency between both models. It synchronizes dependent elements using a synchronization rule, accepting a rule, left and right selectors in the model, and the types of dependent elements. Then, it initiates synchronization using the provided parameters.
+
+
+```csharp
+public class MemberToMember : SynchronizationRule<IFamilyMember, IPerson>
+        {
+            // Synchronization block for a familymember and a person
+            public override void DeclareSynchronization()
+            {
+                Synchronize(m => m.GetFullName(), p => p.Name);
+            }
+        }
+```
+The block defines how individual objects should be synchronized with each other. 
+In this example, people and family members are being synchronized. The synchronization is based on one or more attributes that are the same. A shared attribute between Person and FamilyMember is the attribute `name`, which is why the synchronization is based on the `name` attribute of both classes.
+
+<aside class="note">
+
+**Note:**
+
+At this point, it can be added that when synchronizing between a family and a person register that are already populated, the collision rule applies. For example, there is a man named Smith who marries a woman named Muster. As a result, the woman appears as Smith in the family register and as Muster in the person register. This leads to a collision, and the left side (family register) wins, so the woman's name in the person register is changed to Smith.
```

Comment:
It can be added ist inhaltsleer, sollte entfernt werden. Wichtig ist aber, das passiert nur während der Initialen Synchronisation, nicht während der Updates